src/groth16/mod.rs

```diff
@@ -390,6 +668,46 @@ impl<E: Engine> Parameters<E> {
     }
 }
 
+#[derive(Clone)]
+pub struct MappedParameters<E: Engine> {
```

Comment:
this file is big enough, lets put the `MappedParameters` and implementation details from this PR into its own file

Reply:
Updated.  Let me know if anything else.